### ISSUE TITLE
A reason behind multiplication of criterion with the length of data

### ISSUE THREAD

**Issue by khassanoff** [pytorch/examples#221]
Hi,

Could you please explain, why do you multiply loss by the length of data at evaluation stage?

Regards,
Khassan

PS. Thank you for sharing a great code with the community!

https://github.com/pytorch/examples/blob/3648cbc1a8caa528224750aa2120890fde626fdd/word_language_model/main.py#L116

**Comment by pentiumx** [comment]
Hi, did you figure the reason of this? I have exactly the same question.

**Comment by khassanoff** [comment]
> Hi, did you figure the reason of this? I have exactly the same question.

Hi, yes. If I remember it right, the _nn.CrossEntropyLoss()_ criterion by default takes the average of losses of minibatch elements. Therefore they are multiplying the loss (mean of losses) by the number of elements in the minibatch.

**Comment by ggbioing** [comment]
I don't thing it is correct to measure the loss as a sum of the losses of every observation, especially if we use the loss to select, among different models, the one that performs better.
By absurd, let's consider two perfectly equal models. If the loss depend on the number of observations len(data), we will end up choosing the model tested with more observations with no reason, because in reality they perform the same.

**Comment by khassanoff** [comment]
We usually compare language models using the same evaluation set. There are other requirements for fair comparison such as same training set, same vocabulary, equivalent number of parameters and so on.
